lib/galaxy/tools/expressions/evaluation.py

```diff
@@ -20,11 +22,27 @@
 NODE_ENGINE = os.path.join(FILE_DIRECTORY, "cwlNodeEngine.js")
 
 
-def do_eval(expression: str, jobinput: "CWLObjectType", context: Optional["CWLOutputType"] = None):
+def do_eval(
+    expression: str,
+    jobinput: "CWLObjectType",
+    javascript_requirements: Optional[List[JavascriptRequirement]] = None,
+    outdir: Optional[str] = None,
+    tmpdir: Optional[str] = None,
+    context: Optional["CWLOutputType"] = None,
+):
+    requirements: List[CWLObjectType] = []
```

Comment:
```suggestion
    requirements: List["CWLObjectType"] = []
```

Reply:
Well, I had that before ruff said I need to not do that 😆

Reply:
:stuck_out_tongue_closed_eyes: You may need to tell ruff that Python3.8 is your oldest supported version?

https://typing.python.org/en/latest/guides/modernizing.html#generics-in-the-typing-module

Reply:
I just dropped the type-only import, but good to know

Reply:
Well, you do need the quotes if you had imported it from `if TYPE_CHECKING` and didn't specify `from __future__ import annotations` ; but none of that matters with your proper fix :-)

Reply:
The following runs fine without quotes under Python 3.8:

```python
from typing import (
    List,
    TYPE_CHECKING,
)

from pathlib import Path

if TYPE_CHECKING:
    from pathlib import PurePath


def foo() -> None:
    p: List[PurePath]
    p = [Path("/tmp/foo")]
    print(p[0].as_uri())


foo()
```

Reply:
Sure, that's why I said above that you don't need quotes for type annotation of function variables, only for function parameters and return types. Anyway, that's solved now.